gluefactory/datasets/oxford_paris_mini_1view_jpldd.py

```diff
@@ -39,7 +41,12 @@ class OxfordParisMiniOneViewJPLDD(BaseDataset):
         "seed": 0,
         "num_workers": 0,  # number of workers used by the Dataloader
         "prefetch_factor": None,
-        "reshape": None,  # ex [800, 800]
+        "reshape": None,  # ex [800, 800]  # if reshape is activated AND multiscale learning is activated -> reshape has prevalence
+        "multiscale_learning": {
+            "do": False,
+            "scales_list": [(800, 800), (600, 600), (400, 400)],
+            "scale_selection": 'random' # random or round-robin
+        },
```

Comment:
Gluefactory I think has a preprocessor class that we can directly use maybe? I we reshape to a fixed size we may end up squeezing the image which Remi suggested against doing. We should maintain the aspect ratio ideally i.e. the longer side should be the same after resizing like 512 (as we do in HPatches).

Here is the preprocessor class https://github.com/FelixMoeller3/glue-factory/blob/85180e039fd7f6434b0ee1363189cace3bd9a992/gluefactory/utils/image.py#L12

And here is the usage: https://github.com/FelixMoeller3/glue-factory/blob/85180e039fd7f6434b0ee1363189cace3bd9a992/gluefactory/datasets/hpatches.py#L94

I used this for some other dataloader for POLD2. Let me know if this makes sense and I can help integrate this.

Reply:
I implemented many changes that also reolve this conversation. I will add a comment below explaining in detail